testConnector/README.md

```diff
@@ -0,0 +1,203 @@
+files if configured to do so.
+
+## How to Run the Program
+
+1.  Clone this repository (vantiq-extension-sources) and navigate into `<repo location>/vantiq-extension-sources`.
+2.  Run `./gradlew testConnector:assemble`.
+3.  Navigate to `<repo location>/vantiq-extension-sources/testConnector/build/distributions`. The zip and tar files both 
+contain the same files, so choose whichever you prefer.
+4.  Uncompress the file in the location that you would like to install the program.
+6.  Run `<install location>/testConnector/bin/testConnector` with a local server.config file or specifying the 
+[server config file](#serverConfig) as the first argument. Note that the `server.config` file can be placed in the 
+`<install location>/testConnector/serverConfig/server.config` or `<install location>/testConnector/server.config` 
+locations.
+
+## Logging
+To change the logging settings, edit the logging config file `<install location>/testConnector/src/main/resources/log4j2.xml`,
+which is an [Apache Log4j configuration file.](https://logging.apache.org/log4j/2.x/manual/configuration.html). The logger 
+name for each class is the class's fully qualified class name, *e.g.* "io.vantiq.extjsdk.ExtensionWebSocketClient".  
+
+## Server Config File
+(Please read the [SDK's server config documentation](../extjsdk/README.md#serverConfig) first.)
+
+### Vantiq Options
+*   **authToken**: Required. The authentication token to connect with. These can be obtained from the namespace admin.
+*   **sources**: Required. A comma separated list of the sources to which you wish to connect. Any whitespace will be
+    removed when read.
+*   **targetServer**: Required. The Vantiq server hosting the sources.
+
+# Setting Up Your Vantiq IDE <a name="vantiq" id="vantiq"></a>
+
+## Source Configuration
+
+To set up the Source in the Vantiq IDE, you will need to add a Source to your project. Please check the [Prerequisites](#pre) 
+to make sure you have properly added a Source Definition to your Vantiq namespace. Once this is complete, you can 
+select TestConnector (or whatever you named your Source Definition) as the Source Type. You will then need to fill out 
+the Source Configuration Document.
+
+The Configuration document may look similar to the following example:
+
+    {
+       "testConfig": {
+          "general": {
+             "filenames": ["file1.txt", "file2.txt", "file3.txt"],
+             "pollingInterval": 5000
+          }
+       }
+    }
+
+### Options Available for testConfig
+**Note:** The "testConfig" and "general" portions must be included in the source configuration, but they can be left
+empty.
+*   **filenames**: Optional. A list of String filenames from which to read data. The data will be sent back to Vantiq
+as a notification arriving on the source.
+*   **pollingInterval** Optional. If a list of filenames is provided, a custom polling interval can be specified (in 
```

Comment:
Minor tweak -- but I'd make this required.  If not specified, no polling happens.  `filenames` is an odd way to require polling.  Yes, this requires a code change, but I think it's a better way to do things.